overlord/snapstate/handlers.go

```diff
@@ -522,6 +527,13 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// Restore configuration of the target revision (if available) on revert
+	if snapsup.Revert {
```

Comment:
See the comment note above. I believe this is indeed a requirement, and a test for it would be nice to ensure we're not overlooking that edge case in the future.